hbase-client/src/main/java/org/apache/hadoop/hbase/client/ConnectionUtils.java

```diff
@@ -545,7 +545,7 @@ static void incRegionCountMetrics(ScanMetrics scanMetrics) {
       TableName tableName, Query query, byte[] row, RegionLocateType locateType,
       Function<Integer, CompletableFuture<T>> requestReplica, long rpcTimeoutNs,
       long primaryCallTimeoutNs, Timer retryTimer, Optional<MetricsConnection> metrics) {
-    if (query.getConsistency() == Consistency.STRONG) {
+    if (query.getConsistency() != Consistency.TIMELINE) {
```

Comment:
Whether we will get this issue?
In Query we have
Consistency consistency = Consistency.STRONG
So the getConsistency always will return STRONG if not set right?  Or u mean if some one set like null as consistency?

Reply:
In PE we will set it to null... And we do not prevent users to set it to null so we should take care of this...

Maybe we can file another 'incompatible issue' to not allow users to set it to null, maybe by throwing an exception, but anyway, I do not think we can do this in branch-2 so we still need the patch here...